pkg/utils/utils.go

```diff
@@ -169,6 +177,24 @@ func configSriovDevice(iface *sriovnetworkv1.Interface, ifaceStatus *sriovnetwor
 		if err != nil {
 			glog.Warningf("configSriovDevice(): unable to parse VFs for device %+v %q", iface.PciAddress, err)
 		}
+		// Set Sriov mode for ETH devices
+		if ifaceStatus.Name != "" {
+			if link, err := netlink.LinkByName(ifaceStatus.Name); err == nil && link.Attrs().EncapType == "ether" {
+				for _, vf := range vfs {
+					if err := Unbind(vf); err != nil {
+						glog.Warningf("configSriovDevice(): fail to unbind driver for device %s", vf)
+					}
+				}
+				if err := setNicSriovMode(iface.PciAddress, iface.SriovMode); err != nil {
+					return err
+				}
+				sriovMode, err := GetNicSriovMode(iface.PciAddress)
+				if err != nil {
+					return err
+				}
+				ifaceStatus.SriovMode = sriovMode
```

Comment:
The content of Status shall be update in DiscoverSriovDevices(), this function shall only configure things.

Reply:
PF should have VFs created to be able to read or update the sriov mode, some PFs may not have VFs created, I add a check in the DiscoverSriovDevices() but it will fail if PF doesn't have VFs

Reply:
for PFs that have no VFs, specifying SriovMode should not be allowed (or blocked if SriovMode is switchdev at the very least)

ConfigSriovDevice() starts with configuring VFs, @Mmduh-483  can you elaborate the scenario where updating the ifaceStatus is required here ?

Reply:
agree with @pliurh  the status needs to be updated in one place (DiscoverSriovDevices). its ok imo to have a short period of time where the reflected node state is not in sync with the actual node state.

Reply:
I removed it from the utils it is now updated from DiscoverSriovDevices